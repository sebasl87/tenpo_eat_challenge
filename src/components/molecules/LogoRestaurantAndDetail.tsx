import React from 'react'
import { ScrollView, TouchableOpacity } from 'react-native';
import styled from 'styled-components/native';
import { useNavigation } from '@react-navigation/native';

export interface IRestaurantsLayout {
    image?: any;
    bgColor?: string;
    textNameRestaurant?: string;
}

const Container = styled.View<IRestaurantsLayout>`
    align-items: center;
    background-color: ${(props) => props.bgColor};
    border-radius: 16px;
    display: flex;
    height: 148px;
    justify-content: center;
    margin-top: 10px;
    width: 148px;
    overflow: hidden;
`

const Images = styled.Image`
    width: 148px;
    height: 148px;
`

const ContainerCenter = styled.View`
    width: 100%;
    min-width:100%;
    display: flex;
    justify-content: center;
    flex-direction: row;
    
`

const ContainerDetail = styled.View`
    width: 100%;
    min-width:100%;
    margin-top: 200px;
    display: flex;
    justify-content: center;
    flex-direction: row;
    
`

const TitleRestaurant = styled.Text`
  color: #333333;
  font-size: 22px;
  margin-top: 8px;  
`;

const SubTitleRestaurant = styled.Text`
  color: #adadad;
  font-size: 12px;
  margin-top: 8px;  
`;

const DetailRestaurant = styled.Text`
  color: #adadad;
  font-size: 20px;
  margin-top: 8px;  
`;

const LogoRestaurantAndDetail: React.FC<IRestaurantsLayout> = ({ bgColor, image, textNameRestaurant }) => {
    const navigation = useNavigation()
    return (
        <>
            <ScrollView>
                <ContainerCenter>
                    <TouchableOpacity
                        onPress={() => navigation.navigate("Home")}
                    >
                        <Container bgColor={bgColor}>
                            <Images
                                source={{ uri: image }}
                            />
                        </Container>
                    </TouchableOpacity>
                </ContainerCenter>
                <ContainerCenter>
                    <TitleRestaurant style={{ fontFamily: "Gotham-Bold" }}>{textNameRestaurant}</TitleRestaurant>
                </ContainerCenter>
                <ContainerCenter>
                    <SubTitleRestaurant style={{ fontFamily: "Gotham-Book" }}>Pizzeria - Snacks</SubTitleRestaurant>
                </ContainerCenter>
                <ContainerDetail>
                    <DetailRestaurant style={{ fontFamily: "Gotham-Book" }}>Detalle de restaurante</DetailRestaurant>
                </ContainerDetail>
            </ScrollView>
        </>
    )
};


export default LogoRestaurantAndDetail;