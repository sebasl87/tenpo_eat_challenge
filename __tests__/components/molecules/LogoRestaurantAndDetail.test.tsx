import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import LogoRestaurantAndDetail from "../../../src/components/molecules/LogoRestaurantAndDetail";
import { render, screen, fireEvent } from '@testing-library/react-native';

const onPressMock = jest.fn();

describe("LogoRestaurantAndDetail component", () => {
    it("should render successfully with LogoRestaurantAndDetail", () => {
        const logoRestaurantAndDetail = renderer
            .create(<LogoRestaurantAndDetail textNameRestaurant="LogoRestaurantAndDetail_mock" bgColor="#000" />)
            .toJSON();

        expect(logoRestaurantAndDetail).toMatchSnapshot();
    });

    it("should click On Press", () => {
        render(
            <LogoRestaurantAndDetail textNameRestaurant="LogoRestaurantAndDetail" bgColor="#000" />
        );
        fireEvent.press(screen.getByTestId('clickMe'));
        expect(onPressMock).toBeCalled;
    });
});
