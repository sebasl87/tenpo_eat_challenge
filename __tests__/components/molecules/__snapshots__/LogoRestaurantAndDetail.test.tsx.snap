// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoRestaurantAndDetail component should render successfully with LogoRestaurantAndDetail 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        [
          {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "minWidth": "100%",
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="clickMe"
      >
        <View
          bgColor="#000"
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#000",
                "borderRadius": 16,
                "display": "flex",
                "height": 148,
                "justifyContent": "center",
                "marginTop": 10,
                "overflow": "hidden",
                "width": 148,
              },
            ]
          }
        >
          <Image
            source={
              {
                "uri": undefined,
              }
            }
            style={
              [
                {
                  "height": 148,
                  "width": 148,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "minWidth": "100%",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#333333",
              "fontSize": 22,
              "marginTop": 8,
            },
            {
              "fontFamily": "Gotham-Bold",
            },
          ]
        }
      >
        LogoRestaurantAndDetail_mock
      </Text>
    </View>
    <View
      style={
        [
          {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "minWidth": "100%",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#adadad",
              "fontSize": 12,
              "marginTop": 8,
            },
            {
              "fontFamily": "Gotham-Book",
            },
          ]
        }
      >
        Pizzeria - Snacks
      </Text>
    </View>
    <View
      style={
        [
          {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 200,
            "minWidth": "100%",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#adadad",
              "fontSize": 20,
              "marginTop": 8,
            },
            {
              "fontFamily": "Gotham-Book",
            },
          ]
        }
      >
        Detalle de restaurante
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
